### PR TITLE
ARROW-5856: [Python] Linking 3rdparty cython modules against pyarrow fails since 0.14.0

### PR DESCRIPTION
**Pull request by kszucs**
This is just quick draft for running the cython tests in the wheel builds.

### CODE REVIEW

**Comment by pitrou** [comment]
@ursabot crossbow package -g wheel

File: python/pyarrow/__init__.py [hunk 1]
```diff
@@ -209,12 +209,57 @@ def get_include():
...
+def get_so_version():
+    """
+    Return the SO version for Arrow libraries.
+    """
+    if _sys.platform == 'win32':
+        raise NotImplementedError("Cannot get SO version on Windows")
+    if _has_pkg_config("arrow"):
+        return _read_pkg_config_variable("arrow", ["--variable=so_version"])
+    else:
+        return "100"  # XXX Find a way not to hardcode this?
```
Comment (kszucs) [inline python/pyarrow/__init__.py]: I'll need to cherry pick this for 0.14.1 where we need to use 14

**Comment by pitrou** [comment]
@ursabot crossbow package -g wheel

**Comment by kszucs** [comment]
closing in favor of https://github.com/apache/arrow/commit/690823ce5e863db89d9617818bfe7186d0544cca

**Pull request closed without merging**